src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -327,15 +327,33 @@ internal static RuntimeAssembly InternalLoadAssemblyName(AssemblyName assemblyRe
 
             Debug.Assert(assemblyRef.CodeBase == null);
 
-            return nLoad(assemblyRef, null, ref stackMark, true, assemblyLoadContext);
+            return Load(assemblyRef, null, ref stackMark, true, assemblyLoadContext);
         }
 
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern RuntimeAssembly nLoad(AssemblyName fileName,
-                                                    RuntimeAssembly? assemblyContext,
-                                                    ref StackCrawlMark stackMark,
-                                                    bool throwOnFileNotFound,
-                                                    AssemblyLoadContext? assemblyLoadContext = null);
+
+        [DllImport(RuntimeHelpers.QCall, CharSet = CharSet.Unicode)]
+        private static extern void nLoad(ObjectHandleOnStack fileName,
+                                         ObjectHandleOnStack assemblyContext,
+                                         StackCrawlMarkHandle stackMark,
+                                         bool throwOnFileNotFound,
+                                         ObjectHandleOnStack assemblyLoadContext,
+                                         ObjectHandleOnStack retAssembly);
+
+        private static RuntimeAssembly Load(AssemblyName fileName,
```

Comment:
This does not need to be a separate method. You can just put this inline into InternalLoadAssemblyName .

Reply:
Ok, I have missed that there are two places that call it.